storage/buckets.rb

```diff
@@ -199,6 +199,133 @@ def delete_bucket project_id:, bucket_name:
   # [END delete_bucket]
 end
 
+def set_retention_policy project_id:, bucket_name:, retention_period:
+  # [START storage_set_retention_policy]
+  # project_id       = "Your Google Cloud project ID"
+  # bucket_name      = "Name of your Google Cloud Storage bucket"
+  # retention_period = "Object retention period defined in seconds"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.retention_period = retention_period
+
+  puts "Retention period for #{bucket_name} is now #{retention_period} seconds"
+  # [END storage_set_retention_policy]
+end
+
+def lock_retention_policy project_id:, bucket_name:
+  # [START storage_lock_retention_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Name of your Google Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  # Warning: Once a retention policy is locked it cannot be unlocked
+  # and retention period can only be increased.
+  # Uses bucket.metageneration as a precondition
+  bucket.lock_retention_period!
+
+  puts "Retention policy for #{bucket_name} is now locked"
+  puts "Retention policy effective as of #{bucket.retention_effective_at}"
+  # [END storage_lock_retention_policy]
+end
+
+def remove_retention_policy project_id:, bucket_name:
+  # [START storage_remove_retention_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Name of your Google Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  if !bucket.retention_policy_locked?
+    bucket.retention_period = nil
+    puts "Retention period for #{bucket_name} has been removed."
```

Comment:
Do you want to say "Retention policy for..." instead? You can if you want, per the docs:

> Passing `nil` will remove the retention policy from the bucket if it exists, unless the policy is locked.